aten/src/ATen/Device.h

```diff
@@ -105,6 +105,14 @@ struct Device {
     return type_ == Type::CPU;
   }
 
+  Backend backend() const {
```

Comment:
Let's not do this -- there is no one-to-one function from device to backend and this API is just going to be confusing.  It looks like you just want this for THTensor -- just implement it there.